lib/auth/auth.go

```diff
@@ -2289,6 +2292,10 @@ type newMFADeviceFields struct {
 
 // verifyMFARespAndAddDevice validates MFA register response and on success adds the new MFA device.
 func (a *Server) verifyMFARespAndAddDevice(ctx context.Context, req *newMFADeviceFields) (*types.MFADevice, error) {
+	if len(req.newDeviceName) > mfaDeviceNameMaxLen {
```

Comment:
Is this change backwards compatible? I.e. if there's already MFA device in the backend with >30 character name, will it still work?

Reply:
Yes, this is backwards compatible. It only applies to adding a new device.

Reply:
@codingllama any concerns with limiting the length of MFA devices here?

@kimlisa do you think this warrants a web UI update to prevent users from creating MFA devices with names that are too long?

Reply:
sure i think it's worth it, it's always nice to visually point out to a user where they specifically messed up (highlight red the input box) versus just getting an error message